File: problems/2729-check-if-the-number-is-fascinating/solution.py
class Solution:
    def isFascinating(self, n: int) -> bool:
        s = str(n) + str(2 * n) + str(3 * n)
        if len(s) != 9:
            return False
        digits = set(s)
        if '0' in digits:
            return False
        if len(digits) != 9:
            return False
        for i in range(1, 10):
            if str(i) not in s:
                return False
        return True

File: problems/2729-check-if-the-number-is-fascinating/solution.java
class Solution {
    public boolean isFascinating(int n) {
        String s = Integer.toString(n) + Integer.toString(2 * n) + Integer.toString(3 * n);
        if (s.length() != 9) {
            return false;
        }
        int[] count = new int[10];
        for (int i = 0; i < 9; i++) {
            int digit = s.charAt(i) - '0';
            if (digit == 0) {
                return false;
            }
            count[digit]++;
        }
        for (int i = 1; i <= 9; i++) {
            if (count[i] != 1) {
                return false;
            }
        }
        return true;
    }
}